app/hotels/src/filter/price/PricePopup.js

```diff
@@ -14,105 +14,109 @@ import {
 } from '@kiwicom/mobile-shared';
 import { defaultTokens } from '@kiwicom/mobile-orbit';
 import { SafeAreaView } from 'react-navigation';
+import { Decimal } from 'decimal.js-light';
 
 type Props = {|
   onClose: () => void,
   onSave: ({ minPrice: number, maxPrice: number }) => void,
   isVisible: boolean,
-  min: number,
-  max: number,
-  start: number,
-  end: number,
+  min: Decimal,
+  max: Decimal,
+  start: Decimal,
+  end: Decimal,
   currency: string,
   daysOfStay: number,
 |};
 
-type State = {|
-  price: {
-    start: number,
-    end: number,
-  },
-|};
-
-export default class PricePopup extends React.Component<Props, State> {
-  state = {
-    price: {
-      start: 0,
-      end: 0,
-    },
-  };
+export default function PricePopup({
+  min,
+  max,
+  currency,
+  daysOfStay,
+  onSave,
+  end,
+  start,
+  onClose,
+  isVisible,
+}: Props) {
+  const [price, setPrice] = React.useState({
+    start: new Decimal(0),
+    end: new Decimal(0),
+  });
 
-  static getDerivedStateFromProps = ({ start, end, isVisible }: Props) => {
-    return isVisible
-      ? null
-      : {
-          price: {
-            start,
-            end,
-          },
-        };
-  };
-
-  handlePriceChanged = ([start, end]: number[]) =>
-    this.setState({
-      price: { start, end },
+  React.useEffect(() => {
```

Comment:
So every time `end` and `start` change, this is fired. Isn't it too much with the `handleChange` `useCallback` below?

Reply:
Yeah, this is the problem with destructuring. It is harder to understand what is actually happening. 

the `start` and `end` in the effect is from props. while the `handlePriceChange` below sets the values to state. 

I should refactor this to be more readable.

Reply:
> I should refactor this to be more readable.

So you mean in a future PR or in this PR?

Reply:
I did the refactor already

Reply:
Ah right, forgot to refresh 🙈